### DISCUSSION TITLE
Is it possible to modify network adapters without re-creating the VMs?

### DISCUSSION THREAD

**Discussion by sadoveca** [vagrant-libvirt/vagrant-libvirt#1736] [Q&A]
Hello amazing people, 

I am faced with a scenario where I would like to deploy multiples domains with a requirement to download some packages after deployment. However, after the VMs are up and running the bandwidth needs to be reduced at a minimum (heartbeat only). 
After testing different deployment methods, currently, the most convenient is to provisioning the VMs on an unlimited bandwidth network and then I am manually removing the unlimited network and adding a limited bandwidth interface via: `attach-device VM_NAME interface.xml --persistent`

However, based on the documentation for [vagrant reload](https://developer.hashicorp.com/vagrant/docs/cli/reload), I would expect to be able to adjust the networking interfaces for multiple VMs by reloading a modified Vagrantfile.
Ex. the VM has the management network and a public network named unlimited, then modify the Vagrantfile parameters to completely remove/edit/add another public network, followed by a vagrant reload command. 

Based on the aforementioned documentation: "_After making any modifications to the Vagrantfile, a `reload` should be called._" but in my case this is just rebooting the VMs. 

Am I doing something wrong or does anyone have a better solution for my scenario?

Thanks in advance

**Comment by sadoveca** [comment]
any experienced opinion is welcome

**Comment by oneabrante** [comment]
Starting from the point that there are some specific configurations that we can modify without the need to recreate/shutdown the VMs, we must still remember that there are things we need to do in the VMs just as they should be done on a real host. Vagrant does not offer native functionality to adjust network interfaces at runtime for VMs created with the libvirt provider. In your scenario where you want to initially provision the VMs on an unlimited bandwidth network and then change the network settings later, the approach you are using i.e. manually removing the unlimited network and adding an interface of limited bandwidth via the attach-device command, is the correct way to do this with the libvirt provider. 
As a test attempt, I think it might be interesting to modify the xml of the vm with virsh, for example: sudo virsh edit <domain>, make the changes you deem necessary, then reset with sudo vagrant define </path/domain_xml>. However, it is important to mention that making changes directly to the domain definition XML requires a good understanding of the KVM/libvirt settings and can be an advanced and potentially risky process, as incorrect changes can affect the functioning of the VM. Be sure to back up your domain definition XML before making changes, and carefully test your modifications in test environments to avoid problems.
